Formula/duck.rb

```diff
@@ -20,19 +20,131 @@ class Duck < Formula
 
   depends_on "ant" => :build
   depends_on "maven" => :build
+  depends_on "pkg-config" => :build
   depends_on xcode: :build
 
+  depends_on arch: :x86_64
+  depends_on "libffi"
+  depends_on "openjdk"
+
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "alsa-lib"
+    depends_on "freetype"
+    depends_on "libx11"
+    depends_on "libxext"
+    depends_on "libxi"
+    depends_on "libxrender"
+    depends_on "libxtst"
+
+    ignore_missing_libraries "libjvm.so"
+  end
+
+  resource "jna" do
```

Comment:
This will break when we update the JNA dependency upstream and this formulae will take the older version instead.

Reply:
Thanks for following up on this.  I have 2 thoughts about this:
1) These resources will be made into formulae soon, which will be updated automatically with `livecheck` and `brew bump`.
2) The only thing we're actually taking from JNA is the `dylib/so`, which is built with `-compatibility_version 5` such that it should maintain ABI compatibility as long as the major version remains at 5.  If and when a version 6 is released which breaks ABI compatibility, we'd either rebuild `duck` against version 6 (if it has been updated), or if it takes a while for upstream to update, create a `jna@5` formula to use in the interim.

I'm not intimately familiar with the details of how JNA is implemented, but it's generally rare for native dynamic libraries to break ABI compatibility between minor versions, let alone point releases.  I'm certainly open to being proven wrong on this but we'll have to see how the next update goes.

I should also add here that we are pushing harder to try to avoid downloading pre-built native binaries because doing so has both security and compatibility implications.  However we're hoping to do so without huge inconvenience to upstream projects.  In this particular case, no changes are needed in how the package is built upstream - we simply delete the binaries downloaded by Maven and replace them with ones that will eventually come from formulae.